lib/user-methods.js

```diff
@@ -62,6 +62,19 @@ function checkObjectTypes(obj, types) {
     }
 }
 
+function normalizeSyncUrl(authUrl, syncUrl) {
+    let parsedAuthUrl = new URL(authUrl);
+    let secure = (parsedAuthUrl.protocol === 'https:') ? 's' : '';
+    // Inherit ports from the Auth url
+    let port = (parsedAuthUrl.port === undefined) ? "" : `:${parsedAuthUrl.port}`;
+    let baseUrl = `realm${secure}://${parsedAuthUrl.hostname}${port}`;
+    if (syncUrl) {
+        return new URL(syncUrl, baseUrl, false).toString();
+    } else {
+        return baseUrl + "/default";
```

Comment:
You can avoid the multiple returns if you supply a default value to the `syncUrl`. Something like:

```js
if (!syncUrl) {
    syncUrl = "/default";
}

return new URL(syncUrl, baseUrl, false).toString();
```